File: cpp/practice/Lab06/CircleV6/Circle.cpp
#include "Circle.hpp"

Circle::Circle() {
    _radius = 0;
    _center = new Point();
}

Circle::Circle(double r, Point *c) {
    _radius = r;
    *_center = *c;
}

Circle::Circle(const Circle& c) {
    _radius = c._radius;
    _center = c._center;
}

Circle::~Circle() {
    // nothing to do
}

double Circle::getRadius() const {
    return _radius;
}

Point Circle::getCenter() const {
    return *_center;
}

void Circle::setRadius(double r) {
    _radius = r;
}

void Circle::setCenter(Point *c) {
    *_center = *c;
}

double Circle::getArea() const {
    return PI * _radius * _radius;
}

double Circle::getPerimeter() const {
    return 2 * PI * _radius;
}

string Circle::toString() const {
    stringstream ss;
    ss << "Circle: center = " << _center->toString() << ", radius = " << _radius;
    return ss.str();
}

istream& operator>>(istream& is, Circle* c) {
    cout << "Enter the radius: ";
    is >> c->_radius;
    cout << "Enter the center: ";
    is >> c->_center;
    return is;
}

ostream& operator<<(ostream& os, const Circle* c) {
    os << c->toString();
    return os;
}

File: java/solutions/Lab6/_Circle.java
public class _Circle extends Shape {
    protected double radius = 1.0;

    // methods and constructors
    public _Circle() {
        super();
    }

    public _Circle (double radius, String color, boolean filled) {
        super(color, filled);
        this.radius = radius;
    }

    public double getRadius() {
        return radius;
    }

    public void setRadius (double radius) {
        this.radius = radius;
    }

    public double getArea() {
        return radius * radius * 3.14;
    }

    public double getPerimeter() {
        return 2 * 3.14 * radius;
    }

    @Override
    public String toString() {
        return "Circle | " + super.toString() + " - radius: " + radius + " - Perimeter: " + getPerimeter() + " - Area: " + getArea();
    }
}